osu.Game.Rulesets.Osu/Objects/Drawables/Connections/FollowPointConnection.cs

```diff
@@ -2,54 +2,50 @@
 // See the LICENCE file in the repository root for full licence text.
 
 using System;
+using System.Diagnostics;
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Pooling;
 using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Objects.Pooling;
 using osuTK;
 
 namespace osu.Game.Rulesets.Osu.Objects.Drawables.Connections
 {
     /// <summary>
     /// Visualises the <see cref="FollowPoint"/>s between two <see cref="DrawableOsuHitObject"/>s.
     /// </summary>
-    public class FollowPointConnection : PoolableDrawable
+    public class FollowPointConnection : PoolableDrawableWithLifetime<FollowPointLifetimeEntry>
     {
         // Todo: These shouldn't be constants
         public const int SPACING = 32;
         public const double PREEMPT = 800;
 
-        public FollowPointLifetimeEntry Entry;
         public DrawablePool<FollowPoint> Pool;
 
-        protected override void PrepareForUse()
+        protected override void OnApply(FollowPointLifetimeEntry entry)
         {
-            base.PrepareForUse();
+            base.OnApply(entry);
 
-            Entry.Invalidated += onEntryInvalidated;
-
-            refreshPoints();
+            entry.Invalidated += refreshPoints;
+            refreshPoints(entry);
         }
 
-        protected override void FreeAfterUse()
+        protected override void OnFree(FollowPointLifetimeEntry entry)
         {
-            base.FreeAfterUse();
-
-            Entry.Invalidated -= onEntryInvalidated;
+            base.OnFree(entry);
```

Comment:
Technically an ordering change. The base call nulls out the entry, which used to happen as the last thing in this method. Doesn't seem like it would matter in the context of this class at least, as the entry will still be available via the param, but maybe safer to keep order as it used to be. Not sure.

Reply:
`base.OnFree` is empty for that reason (actual base logic is outside of `OnApply`/`OnFree`). I can remove base calls if you like.

Reply:
Oh never mind, I got confused with this one. Is OK.